### ISSUE TITLE
The accessibility of keyboard navigation of the search bar can be improved

### ISSUE THREAD

**Issue by bick-jp** [internetarchive/openlibrary#7916]
(Not sure if I should report this as a bug. If not, I'll repost this with the more appropriate type of issue.)
The accessibility of keyboard navigation of the search bar can be improved.

1. Currently, the auto-suggested list in the search bar is navigated with the tab key. But it should be navigated via arrow keys. And when the tab key is pressed, the focus should move to the next navigation bar item instead of going into the auto-suggested list.

2. Currently, the auto-suggested list keeps appearing on the screen even after the focus moves out from the search bar. It should disappear when the focus is not on the search bar or on the items of the auto-suggested list.

### Evidence / Screenshot (if possible)
When I input the search word, "design" in this case, it automatically shows suggestions.
![Screen Shot 2023-05-31 at 3 56 58 PM](https://github.com/internetarchive/openlibrary/assets/8709562/d85fba25-baa8-4e24-880f-e6535308e9e4)

When I press the tab key, the focus moves to the search icon.
![Screen Shot 2023-05-31 at 3 57 23 PM](https://github.com/internetarchive/openlibrary/assets/8709562/419fea27-f85d-4048-9d9f-fdcd93034013)

Pressing the tab key again, the focus moves to the barcode-ish icon.
![Screen Shot 2023-05-31 at 3 57 28 PM](https://github.com/internetarchive/openlibrary/assets/8709562/85466ded-297c-40f7-bd7d-e4de6da6a1c5)

Then, the problem is that pressing the tab key again moves the focus to the suggested list.
![Screen Shot 2023-05-31 at 3 57 34 PM](https://github.com/internetarchive/openlibrary/assets/8709562/a6666673-1642-4b15-8e61-f5ba550fe286)

After focusing on all the items in the suggested list, the focus finally moves to the next focusable nev bar item, which is the user icon and hamburger menu in this case.
![Screen Shot 2023-05-31 at 3 57 41 PM](https://github.com/internetarchive/openlibrary/assets/8709562/fa4d1ac4-3929-44c2-b67a-430e10ecd0ec)

Also, another problem is that the suggested list keeps appearing, even after the focus is off from the search bar (in this case, the focus is on "Books").
![Screen Shot 2023-05-31 at 3 58 00 PM](https://github.com/internetarchive/openlibrary/assets/8709562/8c63f30b-55f5-4bfa-b0ea-112f1d6b5114)

### Relevant url?

https://openlibrary.org/search?q=design&mode=printdisabled&has_fulltext=true&subject_facet=Protected+DAISY

### Steps to Reproduce

1. Go to https://openlibrary.org/search?q=design&mode=printdisabled&has_fulltext=true&subject_facet=Protected+DAISY
2. Input any words in the search bar, and navigate the page with the tab key.

* Actual: Explained as above.
* Expected:
** Auto-suggested list should be navigated using arrow keys.
** Pressing the tab key should navigate the focus through the nav bar elements instead of the inside of the auto-suggested list.
** The list should disappear when the focus moves out from the suggested list.
** When the focus is on the list, users should be able to escape fromit by pressing the Esc key, which I think is a suggested accessible behavior.

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** Chrome Version 113.0.5672.126 (Official Build) (arm64)
- **Operating system?** macOS Monterey Version 12.3.1
- **Environment (prod/dev/local)?** prod

### Proposal & Constraints

Search bar keyboard navigation in [Google Scholar](https://scholar.google.com/) and [GitHub](https://github.com/) might be good examples.

**Comment by karlitto01** [comment]
Hi @cdrini. I would like to work on this issue, if possible. Thanks.

**Comment by RayBB** [comment]
@karlitto01 are you still interested in working on this one?

**Comment by RayBB** [comment]
@schu96 you're assigned. Good luck!
